synfig-studio/src/gui/widgets/widget_soundwave.cpp

```diff
@@ -253,6 +253,27 @@ void Widget_SoundWave::on_time_model_changed()
 	queue_draw();
 }
 
+bool Widget_SoundWave::isFileHeaderValid(const std::string &filename)
+{
+	FILE* file = fopen(filename.c_str(), "rb");
+	if (!file)
+		return false;
+	unsigned char header[16];
+	size_t bytesRead = fread(header, 1, 16, file);
+	fclose(file);
+	if (bytesRead < 16)
+		return false;
+	if (header[0]=='R' && header[1]=='I' && header[2]=='F' && header[3]=='F')
```

Comment:
Before each comparison, describe in a comment line what file format(s) are you trying to check.
`// WAV or ... `